presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -137,12 +120,40 @@ public HivePartitionManager(
                 .map(column -> typeManager.getType(column.getTypeSignature()))
                 .collect(toList());
 
-        return partitionColumns.isEmpty() ? ImmutableList.of(new HivePartition(tableName)) : () -> getFilteredPartitionNames(metastore, tableName, partitionColumns, effectivePredicate).stream()
-                // Apply extra filters which could not be done by getFilteredPartitionNames
-                .map(partitionName -> parseValuesAndFilterPartition(tableName, partitionName, partitionColumns, partitionTypes, constraint))
-                .filter(Optional::isPresent)
-                .map(Optional::get)
-                .iterator();
+        TupleDomain<Column> effectivePredicate = convertTupleDomainOfColumnHandleToColumn(effectivePredicateColumnHandles, partitionColumns);
+
+        if (partitionColumns.isEmpty()) {
+            return ImmutableList.of(new HivePartition(tableName));
+        }
+        else {
+            return () -> {
+                List<String> filteredPartitionNames = getFilteredPartitionNames(metastore, tableName, effectivePredicate);
+                return filteredPartitionNames.stream()
+                        // Apply extra filters which could not be done by getFilteredPartitionNames
+                        .map(partitionName -> parseValuesAndFilterPartition(tableName, partitionName, partitionColumns, partitionTypes, constraint))
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .iterator();
+            };
+        }
+    }
+
+    private TupleDomain<Column> convertTupleDomainOfColumnHandleToColumn(TupleDomain<ColumnHandle> effectivePredicateColumnHandles, List<HiveColumnHandle> partitionColumns)
```

Comment:
It does more than a simple convert. How about calling this method as `createPartitionFilterTuppleDomain`